bson-record-codec/src/main/resources/META-INF/native-image/reflect-config.json

```diff
@@ -0,0 +1,5 @@
+[
+{
+  "name":"org.bson.codecs.record.RecordCodecProvider"
```

Comment:
According to https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/Reflection/, `Class.forName("org.bson.codecs.record.RecordCodecProvider")` (it's executed, for example, as a result of calling `Jep395RecordCodecProvider.hasRecordSupport`) requires `org.bson.codecs.record.RecordCodecProvider` to be treated specially. However, we already know that `native-image` is able to detect some things on its own and does not need some reachability metadata produced by the tracing agent. This and similar metadata pieces seem to be not needed, because even without them calling `Jep395RecordCodecProvider.hasRecordSupport` succeeds.

Are you proposing to remove these tree metadata pieces?

P.S. Weirdly, the class initialization report generated by `native-image` does not even mention the `Jep395RecordCodecProvider.CodecProvider` class...

Reply:
The reasons we have reachability metadata for these providers:

```
└── com.mongodb.MongoClientSettings#<clinit>()
    ├── com.mongodb.Jep395RecordCodecProvider#<clinit>()
    │   └── java.lang.Class#forName(java.lang.String) - [ {   "name":"java.lang.Record" }, {   "name":"org.bson.codecs.record.RecordCodecProvider" } ]
    └── com.mongodb.KotlinCodecProvider#<clinit>()
        └── java.lang.Class#forName(java.lang.String) - [ {   "name":"org.bson.codecs.kotlin.DataClassCodecProvider" }, {   "name":"org.bson.codecs.kotlinx.KotlinSerializerCodecProvider" } ]
```